features/nvstore/source/nvstore_int_flash_wrapper.cpp

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+
+// ----------------------------------------------------------- Includes -----------------------------------------------------------
+
+#ifdef DEVICE_FLASH
+
+#include "nvstore_int_flash_wrapper.h"
+#include "nvstore_shared_lock.h"
+#include "FlashIAP.h"
+#include <stdlib.h>
+
+// --------------------------------------------------------- Definitions ----------------------------------------------------------
+
+#define MIN(a,b)            ((a) < (b) ? (a) : (b))
```

Comment:
Is there a std::min and std::max that would be used?

Reply:
Right. Will use std::min and std::max instead.